mpi4jax/cython/mpi_xla_bridge_cpu.pyx

```diff
@@ -69,6 +74,30 @@ cdef void mpi_sendrecv_cpu(void** out_ptr, void** data_ptr) nogil:
     out_ptr[1] = token
 
 
+cdef void mpi_bcast_cpu(void** out_ptr, void** data_ptr) nogil:
+    cdef int32_t nitems = (<int32_t*>(data_ptr[0]))[0]
+    cdef void* send_buf = data_ptr[1]
+    cdef int32_t root = (<int32_t*>(data_ptr[2]))[0]
+    cdef MPI_Comm comm = <MPI_Comm>((<uint64_t*>(data_ptr[3]))[0])
+    cdef MPI_Datatype dtype = <MPI_Datatype>((<uint64_t*>(data_ptr[4]))[0])
+    cdef void* token = data_ptr[5]
+
+    cdef void* out_buf = out_ptr[0]
+
+    cdef int dtype_size, rank
+    ierr = MPI_Type_size(dtype, &dtype_size)
+    abort_on_error(ierr, comm, u"Type_size")
+    ierr = MPI_Comm_rank(comm, &rank)
+    abort_on_error(ierr, comm, u"Comm_rank")
+
+    if rank == root:
+        count = dtype_size * nitems
+        memcpy(out_buf, send_buf, count)
```

Comment:
`out_ptr[0] = send_buf` should do it without `memcpy`

Reply:
isn't it dangerous? The owner off those buffers is XLA and he is managing their lifetime. Nothing guarantees  that you can swap buffers like that. no?

Reply:
Mmh, good point... perhaps it *is* dangerous, but in this case we should also `memcpy` the token, right?

Maybe we can ask the JAX folks about the lifetime of these buffers?

Reply:
I'm pretty confident that in XLA documentation they warn against taking ownership of those buffers. I'll check again.

As or the token, there's no need: it contains garbage data.

Reply:
It could still lead to something like a double free? As in

```
free(in_buf[0])
free(out_buf[0])  // those are the same pointers
```

Reply:
In JAX, they use these idioms:

https://github.com/google/jax/blob/master/jaxlib/lapack.pyx#L83-L84 (CPU)

https://github.com/google/jax/blob/master/jaxlib/cusolver.cc#L189-L193 (GPU)

So it seems that you are right to do the `memcpy`. Still not sure about the token, maybe we should just do it to make sure?

Reply:
I suspect that the token is just random data, but can't confirm. 
For now I'll do what we do in the other kernels (which I really think is wrong, in general, and might be okay only because tokens are never read).